pkg/certificate/mrc_reconciler.go

```diff
@@ -1,214 +1,228 @@
 		}
+
+		log.Error().Err(ErrExpectedActiveMRC).Msgf("expected single MRC with %s intent, found %s", v1alpha2.ActiveIntent, mrc1.Spec.Intent)
+		return nil, nil, ErrExpectedActiveMRC
+	}
+
+	// the combination of active and passive intents is deterministic
+	// regardless of MRC ordering, the passive MRC is the validating MRC and
+	// the active MRC is the signing MRC
+	// the combination of active and active intents is non-deterministic
+	// depending on the MRC ordering, either MRC could be the validating or
+	// signing MRC
+	if intent1 == v1alpha2.ActiveIntent && (intent2 == v1alpha2.PassiveIntent || intent2 == v1alpha2.ActiveIntent) {
+		return mrc1, mrc2, nil
 	}
-	return true
+	if intent1 == v1alpha2.PassiveIntent && intent2 == v1alpha2.ActiveIntent {
+		return mrc2, mrc1, nil
+	}
+
+	log.Error().Err(ErrInvalidMRCIntentCombination).Str(errcode.Kind, errcode.GetErrCodeWithMetric(errcode.ErrInvalidMRCIntentCombination)).
+		Msgf("invalid intent combination of %s and %s", intent1, intent2)
+	return nil, nil, ErrInvalidMRCIntentCombination
 }
 
-// update the MRCComponentStatus
-func (m *Manager) updateMRCComponentStatus(mrc *v1alpha2.MeshRootCertificate) error {
-	// TODO(5046): determine the status to set the owned components to.
-	var status v1alpha2.MeshRootCertificateComponentStatus
-
-	for _, useCase := range m.ownedUseCases {
-		switch useCase {
-		case UseCaseValidatingWebhook:
-			// validating, issuing unknown.
-			mrc.Status.ComponentStatuses.ValidatingWebhook = status
-		case UseCaseMutatingWebhook:
-			mrc.Status.ComponentStatuses.MutatingWebhook = status
-		case UseCaseXDSControlPlane:
-			mrc.Status.ComponentStatuses.XDSControlPlane = status
-		case UseCaseSidecar:
-			mrc.Status.ComponentStatuses.Sidecar = status
-		case UseCaseBootstrap:
-			mrc.Status.ComponentStatuses.Bootstrap = status
-		case UseCaseGateway:
-			mrc.Status.ComponentStatuses.Gateway = status
+func (m *Manager) shouldUpdateIssuers(desiredSigningMRC, desiredValidatingMRC *v1alpha2.MeshRootCertificate) (bool, error) {
+	if desiredSigningMRC == nil || desiredValidatingMRC == nil {
+		log.Error().Err(ErrUnexpectedNilMRC).Msg("unexpected nil MRC provided when getting cert issuers from MRCs")
+		return false, ErrUnexpectedNilMRC
+	}
+
+	m.mu.Lock()
+	signingIssuer := m.signingIssuer
+	validatingIssuer := m.validatingIssuer
+	m.mu.Unlock()
+
+	// if the issuers are already set to the desired value, return
+	if signingIssuer != nil && signingIssuer.ID == desiredSigningMRC.Name && validatingIssuer != nil && validatingIssuer.ID == desiredValidatingMRC.Name {
+		log.Debug().Msgf("issuers already set to the desired value. Will not update issuers: validating[%s] and signing[%s]", validatingIssuer.ID, signingIssuer.ID)
+		return false, nil
+	}
+
+	// if desiredSigningMRC != desiredValidatingMRC and both MRCs have active intents, their state is non
+	// deterministic. To avoid continuously resetting the issuers, only update
+	// if the issuers have not already been set
```

Comment:
This comment applies to the logical NOT of this if statement. Consider rewording so that the comment applies to this if statement.

Reply:
the comment looks to also not be valid anymore?

Reply:
Rewording the comment and refactoring this section to make it more readable.